### PR TITLE
Add translation meta to models and new localization entities

### PR DESCRIPTION
**Pull request by archx3**
####  Summary / Highlights
Introduce `TranslationMeta` class to encapsulate translation fields in various models, enhancing multilingual support. Also, added new models for managing supported languages, community-specific languages, text hashing, and translation caching.

Addresses #1085 , #1080 

#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [x] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

File: src/_main_/utils/base_model.py [hunk 1]
```diff
@@ -25,3 +21,21 @@ def to_json(self, full=False, tiny_info=False):
 
     class Meta:
         abstract = True
+
+class BaseModel(RootModel):
```
Comment (Opoku-Agyemang) [inline src/_main_/utils/base_model.py:25]: i'm a bit concerned about what happens to migrations if we do this.  Does it cause some extra migration files to show up?

File: src/database/models.py [hunk 2]
```diff
@@ -3921,6 +4068,9 @@ class Meta:
         db_table = "feature_flags"
         ordering = ("-name",)
 
+    # class TranslationMeta:
+    #     fields_to_translate = ["name", "scope", "audience", "user_audience", "notes"]
+
```
Comment (Opoku-Agyemang) [inline src/database/models.py:4073]: remove if not needed

File: src/database/models.py [hunk 3]
```diff
@@ -4022,6 +4175,180 @@ def full_json(self):
...
+    ----------
+    name : str
+      name of the language.
+    code : str
+
+    """
+
+    code = models.CharField(max_length=LANG_CODE_STR_LEN, unique=True)
+    name = models.CharField(max_length=SHORT_STR_LEN, unique=True)
+    is_rtl = models.BooleanField(default=False, blank=True) # not used now but maybe used in the future
```
Comment (Opoku-Agyemang) [inline src/database/models.py:4201]: what is rtl?  right to left language? or left to right?

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]
left a few nit comments

File: src/_main_/utils/base_model.py (hunk 1 above)
Comment (archx3) [reply src/_main_/utils/base_model.py:25]: No It doesn't create any extra migration files

File: src/database/models.py (hunk 3 above)
Comment (archx3) [reply src/database/models.py:4201]: Right to left language..
I've changed it to the full meaning to remove any confusion.

**Review by Opoku-Agyemang — APPROVED** [review]
LGTM if it doesn't affect any other migrations.

**Pull request merged**